src/lib.rs

```diff
@@ -1720,6 +1721,8 @@ pub struct OpArgDef {
     type_list_attr: String,
     is_ref: bool,
     // TODO: Add "default_value" and "allowed_values" from OpDef_AttrDef proto
+    default_value: Option<AttrValue>,
+    allowed_values: Option<AttrValue>,
```

Comment:
Hi @adamcrume 
Thank you for your helpful comments.

> The bigger picture here (the issue this PR is addressing) is that we want to support `OpArgDef::default_value` and `allowed_values` but with the more ergonomic Rust-native types instead of the generated proto code, which can be harder to deal with.

So, if I understand this correctly, we want the fields `allowed_values` and `default_values` in `OpDefArg` to be of type `Option<AttrValue>`. However, in `src/lib.rs` `AttrValue` is an enum unlike in protos where it is a struct.

Reply:
Structs with `Option` fields can still derive `Clone`, `PartialEq`, and `Eq`.  Can you explain what you mean?  It does require that `AttrValue` implement those traits, though.

Reply:
When adding the `Clone`, `PartialEq`, and `Eq` traits to the structure the `NameAttrList`, I'm required to apply the same traits to the enum `AttrValue`, but it has fields of type `f32` and `Box` which don't have these traits. I've been stuck here, for a while, now. I will add the cargo check log in a separate gist [here](https://gist.github.com/whereistejas/2da739363c5410ece5414d4f7a3a954e). Am I missing some Rust fundamental about how to assign these traits? Because this becomes an infinite cycle where have I to assign traits to the structs within these structs and the structs within them and so on....